components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v1/impl/core/utils/RecoveryUtil.java

```diff
@@ -0,0 +1,513 @@
+
+        List<RecoveryChannel> recoveryChannelDTOS = new ArrayList<>();
+        if (channels != null) {
+            // Create a response object and add the details to each object.
+            for (NotificationChannelDTO channel : channels) {
+                RecoveryChannel recoveryChannel = new RecoveryChannel();
+                recoveryChannel.setId(Integer.toString(channel.getId()));
+                recoveryChannel.setType(channel.getType());
+                recoveryChannel.setValue(channel.getValue());
+                if (StringUtils.isNotEmpty(channel.getValue())) {
+                    recoveryChannel.setPreferred(channel.isPreferred());
+                }
+                recoveryChannelDTOS.add(recoveryChannel);
+            }
+        }
+        return recoveryChannelDTOS;
+    }
+
+    /**
+     * Handle client errors with specific http codes.
+     *
+     * @param className Class name
+     * @param scenario  Recovery scenario
+     * @param exception IdentityRecoveryClientException
+     * @return WebApplicationException (NOTE: Returns null when the client error is for no user available or for
+     * multiple users available
+     */
+    public static WebApplicationException handleClientExceptions(String className, String tenantDomain,
+                                                                 String scenario, String correlationId,
+                                                                 IdentityRecoveryClientException exception) {
+
+        return handleClientExceptions(className, tenantDomain, scenario, StringUtils.EMPTY, correlationId, exception);
+    }
+
+    /**
+     * Handle client errors with specific http codes.
+     *
+     * @param className Class name
+     * @param scenario  Recovery scenario
+     * @param code      Recovery code
+     * @param exception IdentityRecoveryClientException
+     * @return WebApplicationException (NOTE: Returns null when the client error is for no user available or for
+     * multiple users available
+     */
+    public static WebApplicationException handleClientExceptions(String className, String tenantDomain, String scenario,
+                                                                 String code, String correlationId,
+                                                                 IdentityRecoveryClientException exception) {
+
+        if (StringUtils.isEmpty(exception.getErrorCode())) {
+            return buildConflictRequestResponseObject(className, exception.getMessage(), exception.getErrorCode(),
+                    correlationId);
+        }
+        String errorCode = prependOperationScenarioToErrorCode(exception.getErrorCode(), scenario);
+
+        if (clientErrorMap.containsKey(errorCode)) {
+            String errorCategory = clientErrorMap.get(errorCode);
+
+            // Throw errors according to exception category.
+            switch (errorCategory) {
```

Comment:
this is not required because we can define the mapping from error-code to HTTP status in ErrorMapping.properties file in https://github.com/wso2/identity-rest-dispatcher/blob/master/components/org.wso2.carbon.identity.api.dispatcher/src/main/resources/ErrorMappings.properties

Reply:
`errorCategory` is not related to http error category. The response is different according to the error category. General error type response is not thrown in each case.